### PR TITLE
AssetDetailsPanel stories in new format

### PR DESCRIPTION
**Pull request by amevck**
Need to discuss the issue with getting props documentation automatically.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-503.gearboxjs.preview.cogniteapp.com

File: src/components/AssetDetailsPanel/stories/helper.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { action } from '@storybook/addon-actions';
+import React from 'react';
+import {
+  ASSET_LIST_CHILD,
+  ASSET_ZERO_DEPTH_ARRAY,
+} from '../../../mocks/assetsListV2';
+import { MockCogniteClient } from '../../../utils/mockSdk';
+import { ClientSDKProvider } from '../../ClientSDKProvider';
+
+class CogniteClient extends MockCogniteClient {
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/helper.tsx]: We don't need here such complex implementation here. Better to replace it with something simpler, like:
```typescript
class CogniteClient extends MockCogniteClient {
  assets: any = {
    retrieve: () =>
      new Promise(resolve => {
        setTimeout(() => {
          resolve([fakeAsset]);
        }, 1000);
      }),
  };
}
```

File: src/components/AssetDetailsPanel/stories/helper.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+export const decorators = [
+  (storyFn: any) => (
+    <ClientSDKProvider client={client}>{storyFn()}</ClientSDKProvider>
+  ),
+];
+
+export const onAssetLoaded = (asset: Asset) =>{
+  console.log(asset)
+action('handleAssetLoaded')(asset)
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/helper.tsx]: we will not see the result of this action cause action tab is unavailable for Docs view

Comment (amevck) [reply src/components/AssetDetailsPanel/stories/helper.tsx]: Roman action tab unavailable for docs but it is shown in the canvas tab.

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-503.gearboxjs.preview.cogniteapp.com

File: src/components/AssetDetailsPanel/stories/basic.story.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { Meta, Story, Preview, Props} from '@storybook/addon-docs/blocks';
+import { AssetDetailsPanel } from '../AssetDetailsPanel';
+import {AssetDetailsPanelPure} from '../AssetDetailsPanelPure';
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/basic.story.mdx]: you need to stay consistent in formatting

File: src/components/AssetDetailsPanel/stories/basic.story.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# Asset Details Panel
+
+This component loads meta information for an asset and presents it as a key/value list in a table with two columns.
+The component is used as one of the panes in AssetMeta component.
+
+**NOTE:** The component should have `ClientSDKProvider` as a parent component in react component tree.
+
+## Basic usage:
+
+Don't forget to imports:
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/basic.story.mdx]: Replace this with only one word – `Imports:`

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import { action } from '@storybook/addon-actions';
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: you don't need this import here

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
+import { AssetDetailsPanel } from '../AssetDetailsPanel';
+import { decorators,  customElementRendering, handleAssetLoaded, onAssetLoaded, toCategory, customStyle} from './helper';
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: `customElementRendering, handleAssetLoaded` are not used here

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+## Custom categories
+
+
+```jsx
+const toCategory = (name: string): string => name.split('_')[0];
+```
+
+<Preview>
+  <Story name="Custom categories">
+        <AssetDetailsPanel
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: Formatting

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    />
+  </Story>
+</Preview>
+
+## Custom Styles
+
+```jsx
+import CSS from 'csstype';
+
+//...
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: Better to remove it

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+const customStyle: CSS.Properties = {
+  border: '1px solid red',
+};
+```
+
+<Preview>
+  <Story name="Custom Styles">  
+  <AssetDetailsPanel
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: Formatting

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+</Preview>
+
+## Handle callbacks
+
+A callback function can be provided to fire when the asset is loaded:
+
+```jsx
+import { Asset } from '@cognite/sdk';
+
+// ...
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: Remove this

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+const onAssetLoaded = (asset: Asset) => console.log(asset);
+```
+
+<Preview>
+  <Story name="Load Callback">  
+  <AssetDetailsPanel
+      assetId={4650652196144007}
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: Formatting

File: src/components/AssetDetailsPanel/stories/helper.tsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+export const customStyle: CSS.Properties = {
+  border: '1px solid red',
+};
+
+export const toCategory = (name: string): string => name.split('_')[0];
+
+export const onAssetLoaded = (asset: Asset) => {
+  console.log(asset);
+  action('handleAssetLoaded')(asset);
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/helper.tsx]: Remove this, cause actions are not available for docs tab

File: src/components/AssetDetailsPanel/stories/helper.tsx [hunk 13]
```diff
@@ -0,0 +1,37 @@
+import { Asset } from '@cognite/sdk';
+import { action } from '@storybook/addon-actions';
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/helper.tsx]: Remove this

File: src/components/AssetDetailsPanel/stories/examples.story.mdx [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+      customSpinner={<div>Loading...</div>}
+    />
+  </Story>
+</Preview>
+
+## Custom Styles
+
+```jsx
+import CSS from 'csstype';
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/examples.story.mdx]: That's wrong.  Check the type of `customStyle` prop in code (`AssetDetailsPanelPure.tsx` file).

File: src/components/AssetDetailsPanel/stories/helper.tsx [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+import CSS from 'csstype';
```
Comment (mrtraser) [inline src/components/AssetDetailsPanel/stories/helper.tsx]: This is a wrong package to import CSS type from.

File: src/components/AssetDetailsPanel/stories/helper.tsx (hunk 13 above)
Comment (amevck) [reply src/components/AssetDetailsPanel/stories/helper.tsx]: export const onAssetLoaded = (asset: Asset) => {
  console.log(asset);
}

Roman this call-back methods need Asset?

Comment (mrtraser) [reply src/components/AssetDetailsPanel/stories/helper.tsx]: I meant that you need to remove line 2 here

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-503.gearboxjs.preview.cogniteapp.com

**Pull request merged**